### PR TITLE
fix #1134 assigned groups while in user manager

### LINKED ISSUE #1134: Assign groups do not work in user manager
When I try to assign a group to a user, this operation is not applied. 
Also enabled checkbox can not be updated.

### PR DESCRIPTION
**Pull request by saidaipparla**
changes  taken place 
1. changed the json format for server side both for creation and updating user 
2. get the username and update the view when user edits

### CODE REVIEW

File: web/client/actions/users.js [hunk 1]
```diff
@@ -236,23 +236,33 @@ function createError(user, error) {
...
             }).catch((error) => {
                 dispatch(saveError(user, error));
             });
         }
         // createUser
         dispatch(creatingUser(user));
-        return API.createUser(user, options).then((id) => {
+        let userToPost = {...user};
+        if (user && user.groups) {
+            userToPost = {...user, groups: { group: user.groups.filter((g) => {
```
Comment (offtherailz) [inline web/client/actions/users.js]: Add a comment here referencing the issue on geostore

File: web/client/actions/users.js [hunk 2]
```diff
@@ -236,23 +236,33 @@ function createError(user, error) {
...
 
 function saveUser(user, options = {}) {
-    return (dispatch) => {
+    return (dispatch, getState) => {
         if (user && user.id) {
             dispatch(savingUser(user));
-            return API.updateUser(user.id, {...user, group: user.groups}, options).then((userDetails) => {
+            return API.updateUser(user.id, {...user, groups: { group: user.groups}}, options).then((userDetails) => {
                 dispatch(savedUser(userDetails));
+                let state = getState && getState();
```
Comment (offtherailz) [inline web/client/actions/users.js]: This is wrong, use the previous search text

**Comment by offtherailz** [comment]
going to close this pull request, redo it from your fork

File: web/client/actions/users.js (hunk 1 above)
Comment (saidaipparla) [inline web/client/actions/users.js]: https://github.com/geosolutions-it/geostore/issues/149

**Comment by saidaipparla** [comment]
ok i will do it

**Pull request closed without merging**